lib/colorado_lottery.rb

```diff
@@ -0,0 +1,60 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants, :winners, :current_contestants
+  def initialize
+    @registered_contestants = Hash.new { |hash, key| hash[key] = [] }
+    @winners = []
+    @current_contestants = Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def interested_and_18?(contestant, game)
+    contestant.game_interests.include?(game.name) && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    interested_and_18?(contestant, game) &&
+    (!contestant.out_of_state? || game.national_drawing?)
+  end
+
+  def register_contestant(contestant, game)
+
+    registered_contestants[game.name] << contestant if can_register?(contestant,game)
+  end
+
+  def eligible_contestants(game)
+    registered_contestants[game.name]
```

Comment:
This worked, because of the way the interaction pattern was set up, but it should have included a `find_all` to ensure all contestants had greater than or equal spending money to the `game.cost`.